### ISSUE TITLE
Issue with handling of emoji

### ISSUE THREAD

**Issue by sloriot** [doxygen/doxygen#6612]
I observe some new warnings and some emoji replacing namespaces in class name.

For example:
```
models:2: warning: Unsupported emoji ':Surface_mesh_parameterization:' found while handling command \ref
```
With the following output in the name of a class:
![screenshot from 2018-11-12 14-37-36](https://user-images.githubusercontent.com/1224632/48350784-879db300-e688-11e8-9991-a0545c90eb42.png)

The warnings are produced for exemple by [this line](https://github.com/CGAL/cgal/blob/master/Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/Fixed_border_parameterizer_3.h#L77) by:
```
/// \cgalModels `Parameterizer_3` 
```

with:
```
"cgalModels=\xrefitem models \"Is Model Of\" \"Is Model Relationships\"" \
```

Note that even if removing the lines with `\cgalModels` remove the warning, the thumbup emoji is still present in the doc.

Steps to reproduce the issue:
```
git clone git@github.com:CGAL/cgal.git
mkdir doc_build && cd doc_build
cmake -DDOXYGEN_EXECUTABLE=/PATH/TO/DOXYGEN/EXE/doxygen ../cgal/Documentation/doc/
make Documentation_doc
make Surface_mesh_parameterization_doc
firefox doc_output/Surface_mesh_parameterization/classCGAL_1_1Surface__mesh__parameterization_1_1Fixed__border__parameterizer__3.html
```

**Comment by albert-github** [comment]
Looking into the issue, found already some problems
- the emoji as shown above
- majority of `\ref` errors (and subsequent error messages about `TK_...`)

all where problems due to the fact that `:` was caught and subsequently the next `:` was seen as begin of an emoji. Catching now `::` properly.

@sloriot Getting warnings about configuration parameter: `EXTRACT_ALL_NO_DETAILED_IF_EMPTY`, how to get rid of it? Is this a special Cmake setting ?

**Comment by sloriot** [comment]
You can ignore it, it is a special config flag from a patch that Dimitri did not want to integrate upstream, see #452.

**Comment by albert-github** [comment]
Wouldn't it be better to remove the flag? or are you running with a version of doxygen where you have this flag incorporated?

**Comment by sloriot** [comment]
I'm waiting for the feature Dimitri mentioned in the discussion to remove it. In the meantime, we maintain release branches of doxygen in which we merge the rejected PR (because it is less complicated than working around the issue for now).

**Comment by sloriot** [comment]
Indeed, fixed. Thanks!

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #6614

**Comment by sloriot** [comment]
I gave it a try locally and warnings as well as thumbup are gone. Thanks!

**Comment by albert-github** [comment]
Code has been integrated in master on github.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.15.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
